### PR TITLE
added iterative example

### PR DESCRIPTION
**Pull request by prix0007**
Adds an Iterative example in the enums section.

### CODE REVIEW

**Comment by enitrat** [comment]
Can you just add some context above the example and say that we demonstrate two different ways, using a) recursive functions and b) loops?

Thank you for this contribution!

**Comment by prix0007** [comment]
Makes sense.
Thanks for mentioning, updated accordingly

File: src/ch05-01-enums.md [hunk 1]
```diff
@@ -95,6 +95,10 @@ The `Option` enum is helpful because it allows you to explicitly represent the p
 
 To give you an example, here is a function which returns the index of the first element of an array with a given value, or None if the element is not present.
 
+We are demonstrating two approaches for above function:
```
Comment (enitrat) [inline src/ch05-01-enums.md:98]: typo: for the above function

**Review by enitrat — APPROVED** [review]

File: src/ch05-01-enums.md (hunk 1 above)
Comment (prix0007) [reply src/ch05-01-enums.md]: fixed, thanks

**Pull request merged**